build/Dockerfile

```diff
@@ -1,46 +1,289 @@
+	make modules
+
+FROM debian AS tracer-downloader
+
+ARG JAEGER_VERSION=v0.4.2
+
+RUN set -x \
+	&& apt-get update \
+	&& apt-get install --no-install-recommends --no-install-suggests -y ca-certificates apt-transport-https wget \
+	&& wget https://github.com/jaegertracing/jaeger-client-cpp/releases/download/${JAEGER_VERSION}/libjaegertracing_plugin.linux_amd64.so -O /usr/local/lib/libjaegertracing_plugin.so
+
+
+FROM debian as opentracing
+ARG NGINX_VERSION
+ARG OPENTRACING_VERSION=1.6.0
+
+COPY --from=opentracing-builder /nginx-${NGINX_VERSION}/objs/ngx_http_opentracing_module.so /usr/lib/nginx/modules/ngx_http_opentracing_module.so
+COPY --from=opentracing-builder /usr/local/lib/libopentracing.so.${OPENTRACING_VERSION} /usr/local/lib/libopentracing.so.1
+COPY --from=tracer-downloader /usr/local/lib/libjaegertracing_plugin.so /usr/local/lib/libjaegertracing_plugin.so
+
+RUN ldconfig
+
+############################################# Build image for Opentracing with NGINX Plus #############################################
+FROM debian-plus as opentracing-plus
+
+ENV NGINX_OPENTRACING_MODULE_VERSION 23+0.9.0-1~buster
+
+RUN --mount=type=secret,id=nginx-repo.crt,dst=/etc/ssl/nginx/nginx-repo.crt,mode=0644 \
+	--mount=type=secret,id=nginx-repo.key,dst=/etc/ssl/nginx/nginx-repo.key,mode=0644 \
+	apt-get update && \
+	apt-get install --no-install-recommends --no-install-suggests nginx-plus-module-opentracing=${NGINX_OPENTRACING_MODULE_VERSION} && \
+	rm -rf /var/lib/apt/lists/*
+
+COPY --from=tracer-downloader /usr/local/lib/libjaegertracing_plugin.so /usr/local/lib/libjaegertracing_plugin.so
+
+############################################# Cleanup for Debian with NGINX Plus and App Protect #############################################
+FROM debian-plus-ap-intermediate as debian-plus-ap
+RUN rm /etc/apt/apt.conf.d/90app-protect-sigs /etc/apt/sources.list.d/app-protect-sigs.list \
+	&& mkdir -p /etc/nginx/waf /etc/nginx/waf/nac-policies /etc/nginx/waf/nac-logconfs /etc/nginx/waf/nac-usersigs /var/log/app_protect /opt/app_protect \
+	&& chown -R nginx:0 /etc/app_protect /usr/share/ts /var/log/app_protect/ /opt/app_protect/ /var/log/nginx/ \
+	&& touch /etc/nginx/waf/nac-usersigs/index.conf
+
+RUN printf "MODULE = ALL;\nLOG_LEVEL = TS_CRIT;\nFILE = 2;\n" > /etc/app_protect/bd/logger.cfg \
+	&& printf "[config_set_compiler]\nlog_level=fatal\n" >> /etc/app_protect/tools/asm_logging.conf \
+	&& for v in \
+	asm_config_server \
+	lock_factory \
+	bd_agent \
+	import_export_policy \
+	set_active \
+	; do sed -i "/\[$v/a log_level=fatal" "/etc/app_protect/tools/asm_logging.conf" \
+	; done
+
+COPY --chown=nginx:0 build/log-default.json /etc/nginx
+
+
+############################################# Create common files, permissions and setcap #############################################
+FROM $BUILD_OS as files
+
+ARG PLUS
+
+RUN mkdir -p /var/lib/nginx /etc/nginx/secrets /etc/nginx/stream-conf.d /etc/nginx/oidc \
```

Comment:
for readability, would it make sense to create  each file on a separate line like we had before https://github.com/nginxinc/kubernetes-ingress/blob/master/build/DockerfileForPlus#L55 ?